From 599032d09f090127b756ea440a56110f617aa317 Mon Sep 17 00:00:00 2001
From: Daniel Johnson <hexahedria@gmail.com>
Date: Thu, 5 Nov 2015 18:18:56 -0800
Subject: [PATCH 07/10] Change dumping mechanism to use non-secure mode

---
 target-arm/translate.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/target-arm/translate.c b/target-arm/translate.c
index 5038ae5..61ce88e 100644
--- a/target-arm/translate.c
+++ b/target-arm/translate.c
@@ -11561,9 +11561,11 @@ unsigned int get_spsr_in_mode(CPUARMState *env, int mode) {
 
 unsigned int get_coprocessor_reg(ARMCPU *cpu, int crn) {
     CPUARMState *env = &cpu->env;
-    const ARMCPRegInfo *ri = get_arm_cp_reginfo(cpu->cp_regs, ENCODE_CP_REG(15,0,0,crn,0,0,0));
+    const ARMCPRegInfo *ri = get_arm_cp_reginfo(cpu->cp_regs, ENCODE_CP_REG(15,0,1,crn,0,0,0));
+    unsigned int res = read_raw_cp_reg(env, ri);
+    // printf("ARMCPURegInfo name %s, type %i, crn %i     -> %x\n", ri->name, ri->type, ri->crn, res);
 
-    return read_raw_cp_reg(env, ri);
+    return res;
 }
 
 void arm_cpu_dump_state(CPUState *cs, FILE *f, fprintf_function cpu_fprintf,
-- 
1.7.1

